File: web/src/mocks/handlers/info.ts
/**
 * MSW handlers for app info API endpoint
 */

import { delay, HttpResponse, http } from "msw";
import type { components } from "@/types/api.generated";

type AppInfoDto = components["schemas"]["AppInfoDto"];

export const infoHandlers = [
  // App info endpoint (public, no authentication required)
  http.get("/api/v1/info", async () => {
    await delay(50);
    const info: AppInfoDto = {
      version: "1.0.0-mock",
      name: "codex",
    };
    return HttpResponse.json(info);
  }),
];

File: crates/codex-api/src/routes/v1/handlers/info.rs
//! Application info handler

use std::sync::Arc;

use axum::{Json, extract::State};

use super::super::dto::AppInfoDto;
use crate::extractors::AppState;

/// Get application information
///
/// Returns the application name and version.
/// This endpoint is public (no authentication required).
#[utoipa::path(
    get,
    path = "/api/v1/info",
    responses(
        (status = 200, description = "Application info", body = AppInfoDto),
    ),
    tag = "Info"
)]
pub async fn get_app_info(State(state): State<Arc<AppState>>) -> Json<AppInfoDto> {
    Json(AppInfoDto {
        version: state.app_version.to_string(),
        name: state.app_name.to_string(),
    })
}
